metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -630,8 +650,10 @@ class MetalsLanguageServer(
     savedFiles.add(path)
     // read file from disk, we only remove files from buffers on didClose.
     buffers.put(path, path.toInput.text)
-    trees.didChange(path)
-    onChange(List(path))
+    Future
+      .sequence(List(parseTrees(path), onChange(List(path))))
```

Comment:
`SyntaxErrorSlowSuite` had become flaky until we merged the `parseTrees` and `onChange` futures here